beard/metrics/text.py

```diff
@@ -0,0 +1,170 @@
+            for j in s2_occur:
+                if i - H <= j <= i + H:
+                    if j not in s2_matched_positions:
+                        matches += 1
+                        s2_matched_positions.append(j)
+                    s1_matching_letter.append((i, letter))
+                    break
+
+    for letter, (s1_occur, s2_occur) in letters_cache.items():
+        for j in s2_occur:
+            for i in s1_occur:
+                if i - H <= j <= i + H:
+                    if i not in s1_matched_positions:
+                        s1_matched_positions.append(i)
+                    s2_matching_letter.append((j, letter))
+                    break
+
+    s1_matching_letter.sort()
+    s2_matching_letter.sort()
+    transpositions = len(tuple(filter(lambda x: x[0][1] != x[1][1],
+                               zip(s1_matching_letter,
+                                   s2_matching_letter))))
+    return matches, transpositions
+
+
+def jaro(s1, s2):
+    """Return the Jaro similarity of the strings s1 and s2.
+
+    Parameters
+    ----------
+    :param s1: string
+        First string
+
+    :param s2: string
+        Second string
+
+    Returns
+    -------
+    :returns: float
+        Similarity of s1 and s2
+
+    Reference
+    ---------
+    Jaro, M. A. 1989. Advances in record-linkage methodology as
+    applied to matching the 1985 census of Tampa, Florida.
+    Journal of the American Statistical Association
+    84:414–420.
+
+    Jaro, M. A. 1995. Probabilistic linkage of large public health data
+    files (disc: P687-689).
+    Statistics in Medicine
+    14:491–498.
+
+    Winkler, W. E. 1999. The state of record linkage and cur-
+    rent research problems. Statistics of Income Division, In-
+    ternal Revenue Service Publication R99/04. Available from
+    http://www.census.gov/srd/www/byname.html.
+    """
+    n_matches, n_transpositions = _jaro_matching(s1, s2)
+
+    return (1/3)*((n_matches/len(s1)) +
+                  (n_matches/len(s2)) +
+                  ((n_matches - (n_transpositions/2))/(n_matches)))
+
+
+def jaro_winkler(s1, s2, p = 0.1):
+    """Return the Jaro-Winkler similarity of the strings s1 and s2.
+
+    Parameters
+    ----------
+    :param s1: string
+        First string
+
+    :param s2: string
+        Second string
+
+    Returns
+    -------
+    :returns: float
+        Similarity of s1 and s2
+
+    Reference
+    ---------
+    Winkler, W. E. 1999. The state of record linkage and cur-
+    rent research problems. Statistics of Income Division, In-
+    ternal Revenue Service Publication R99/04. Available from
+    http://www.census.gov/srd/www/byname.html.
+    """
+    jaro_distance = jaro(s1, s2)
+
+    common_prefix = 0
+    for s1_letter, s2_letter in zip(s1, s2):
+        if s1_letter == s2_letter and common_prefix < 4:
+            common_prefix += 1
+        else:
+            break
+
+    return jaro_distance + (p * min(common_prefix, 4) * (1 - jaro_distance))
```

Comment:
Now that the loop breaks early, you can directly use `common_prefix` instead of `min(...)`